src/sentry/integrations/utils/code_mapping.py

```diff
@@ -554,6 +509,12 @@ def find_roots(stack_path: str, source_path: str) -> tuple[str, str]:
 
             if stack_root:  # append trailing slash
                 stack_root = f"{stack_root}{stack_path_delim}"
+            if source_root and source_root[-1] != SLASH:
+                source_root = f"{source_root}{SLASH}"
+            if stack_path.endswith(".py") and len(overlap_to_check) > 1:
+                next_dir = overlap_to_check[0]
```

Comment:
To imagine how this new logic works, imagine we have the following:

- Stack trace: `"sentry/api/endpoints/project_stacktrace_link.py"`
- Source path: `"api/endpoints/project_stacktrace_link.py"`

First we split them into their "items" and start :

- Stack trace items (`overlap_to_check`): `["sentry", "api", "endpoints", "project_stacktrace_link.py"]`
- Source path items: `["api", "endpoints", "project_stacktrace_link.py"]`

Then we join the overlap and check if it is a trailing substring of the source path.  Here we check if `"sentry/api/endpoints/project_stacktrace_link.py"` is a trailing substring of `"api/endpoints/project_stacktrace_link.py"`. It is not, so we pop the first element off of the overlap to check, and add it to the current list of stack root items. Here we pop `"sentry"`. 

Now repeat. We once again check if the overlap to check is a trailing substring of the source path. Now we are checking if `"api/endpoints/project_stacktrace_link.py"` is a trailing substring of `"api/endpoints/project_stacktrace_link.py"`. Since it is, we can now construct the roots. 

We first derive the **source code root** by taking everything before the overlap. In this case it is just `""`. Then we derive the **stack trace root** by taking merging the items we popped off the stack trace path. In this case just `sentry`. We then apply some standardization like adding a trailing slash.

This is how we arrive at the correct solution of:

- **Stack trace root: ** `"sentry/"`
- **Source code root: ** `""`